### PR TITLE
Add new IDL for optionals, final, appendable and mutables [19261]

### PR DESCRIPTION
**Pull request by richiware**

### CODE REVIEW

File: IDL/optional.idl
Comment (adriancampo) [inline IDL/optional.idl:1]: Maybe it is worth considering splitting the @optional tests into multiple IDL files.

**Review by adriancampo — COMMENTED** [review]
Should optionals, final, appendable, and mutables be added to inheritance.idl and declarations.idl?

What about container types?
Example:
struct MapShortInnerMutableStructureHelper
{
    map<short, InnerMutableStructureHelper> var_map_short_innermutablestructurehelper;
};

File: IDL/optional.idl
Comment (richiware) [reply IDL/optional.idl:1]: Right now the number of tests are not enough to split it, in my opinion.

**Comment by richiware** [comment]
Added inheritance as suggested. Forward declarations don't impact in @mutable, etc..

**Review by adriancampo — APPROVED** [review]
LGTM

**Pull request merged**